jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -79,6 +80,9 @@ def process(self, input_data):
         self.log.info(f"Outlier Detection mode: {mode}")
 
         # determine the asn_id (if not set by the pipeline)
+        if mode == "imaging":
```

Comment:
I'm confused by this comment. As written, for a `str` input, the library does not open any models.  The function `_get_asn_id()` doesn't open any models if it receives a `ModelLibrary`, and the conditional on line 83 that you highlighted ensures that for imaging data that function always does receive a model library.

I don't understand how you would be able to move the assignment of the asn_id into `imaging.detect_outliers` because it takes that as input

Reply:
Hopefully I can describe what's going on in my head :) If not, let me know and I can open up a PR with the proposed changes.

The goal is to remove this `if mode == "imaging"` check. To accomplish this I think we could update `_get_asn_id` to read the `asn_id` directly from the input association (looking at it now I don't think it even needs to make a `ModelLibrary` that can be done within `imaging.detect_outliers`). This is only used/important when `OutlierDetectionStep` is called directly with an association filename as an input.

The main justification for this is to keep the modes as separate as possible. There's already a `mode == "imaging"` check below and `imaging.detect_outliers` already contains to create a `ModelLibrary` from an association. I think removing the check here would simplify the code a bit.

Reply:
Thanks!